File: Backend/login-microsoft365/src/main/java/com/miapp/model/Programa.java
package com.miapp.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "PROGRAMA")
public class Programa {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "IDPROGRAMA")
    private Long idPrograma;

    @Column(name = "PROGRAMA", length = 20, nullable = false)
    private String programa;

    @Column(name = "DESCRIPCION_PROGRAMA", length = 150, nullable = false)
    private String descripcionPrograma;

    @Column(name = "ACTIVO", columnDefinition = "NUMBER(1)")
    private Boolean activo = true;
}

File: Frontend/sakai-ng-master/src/app/biblioteca/interfaces/programa.ts
export class Programa {
    id: number;
    programa: string;
    descripcionPrograma: string;
    activo: boolean;
    constructor(init?: Partial<Programa>) {
        this.id = 0;
        this.programa = '';
        this.descripcionPrograma = '';
        this.activo = true;
        Object.assign(this, init);
    }
}
